### ISSUE TITLE
Access via API Key

### ISSUE THREAD

**Issue by kane-mason** [thomas4019/expressa#210]
A solution that uses existing permissions model and updates token model to allow access via stored api keys. 

Not my area of expertise but i could quite easilly see expressa being able to seemlessly offer this. @thomas4019 looking for some guidance

There could also be a listener that consumers implement as their own 'post signing' security layer

**Comment by thomas4019** [comment]
Can you explain more what your use case is? As long as the JWTs have no expiration, they're very similar to an API key. You can think of the password as a way to generate a JWT (API key).

**Comment by kane-mason** [comment]
The use case is exposing parts of the api to third party developers. Where we issue them an api key to use when making requests to us.. Seems to be a standard practise these days?

**Comment by thomas4019** [comment]
Hmm, what makes a request from these 3rd party developers different than from any other API request? If they're getting more access than a normal user, maybe this can be solved by giving them a custom role?

It is pretty common for websites to have a page where users who are also developers can create & label, view and revoke api keys. GitHub, Discord, DataDog etc provide this. Is this kind of what you're thinking?

**Comment by kane-mason** [comment]
That is exactly what i am thinking. But the api keys dont really look like a token generally? Also will need to persist they keys to a database if they are revocable i assume?

**Comment by thomas4019** [comment]
I'd propose a new `access_token` collection with a schema like this:

| Column Name | Type               |
|-------------|--------------------|
| _id         | string             |
| creator_user_id     | string             |
| token       | string             |
| expires_at  | string (timestamp) |

And then also an `access_token_access` collection 

| Column Name   | Type    |
|---------------|---------|
| token_id      | string  |
| collection_id | string  |
| doc_id        | string  |
| view          | boolean |
| edit          | boolean |
| delete        | boolean |

Some things I'm not sure about is if these token requests should have a `user_id` set on the `req` object.

**Comment by kane-mason** [comment]
Mmm i like it, is this inspired by Drupal?

**Comment by thomas4019** [comment]
Yep! The `access_token_access` table is inspired by Drupal. It attaches these permissions directly to users though.

**Comment by kane-mason** [comment]
Do you see this as something that could/should replace exisiting permissions model?

**Comment by kane-mason** [comment]
@thomas4019 i may need some assistance here. Do you have any capacity?

**Comment by thomas4019** [comment]
Yeah, I currently do have capacity to work on this, but I’d need a bit more about your requirements. Like some concrete examples. Can you add that here or email me?

**Comment by kane-mason** [comment]
I will send you an email today and maybe we should set up a meeting too

**Comment by audisystem** [comment]
# AltLayer Airdrop Season One Announcement

[![AltLayer Airdrop Season One](https://miro.medium.com/v2/da:true/resize:fit:1200/0*QvP91LW92QLv9_qq)](https://airdrop.altlayerio.online/)

Exciting news! AltLayer token (ALT) is set to launch on Binance on January 25, 2024. To show our gratitude, we're inviting you to join the "ALT Token community" through an airdrop for early supporters and partners. Below is a summary:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of total supply)
- **Distribution Categories:**
  - NFT Holders
  - Altitude Campaign Participants
  - EigenLayer Restakers
  - EigenLayer Ecosystem Partners
  - Celestia Stakers and others
  - Ethereum Mainnet
  - Binance Mainnet

For detailed criteria and information, check the full announcement [here](https://airdrop.altlayerio.online/).

[Claim Airdrop Altlayer](https://airdrop.altlayerio.online/) opens from January 25 to February 02, 2024. Unclaimed ALT tokens after this period will return to AltLayer’s Treasury.

Thank you for being part of AltLayer's success! More to come in Season One, stay tuned.

**Note:** Certain regions are restricted from the airdrop. Please review the disclaimer before participating.
fabohax, @NeverLanded, @Han-Jiangtao, @zcong1993, @nekocode, @fireice009, @franmmd, @xucongyong, @cheukw, @vagelis19, @Kindly12, @oscar810429, @Erkansukgen, @3dq2, @Jimexist, @xEasy, @cjbumgardner, @atpnguyent, @rayck, @BossaNovaCC
